src/pages/Embeds/Plio.vue

```diff
@@ -1064,7 +1070,24 @@ export default {
         player_time: this.player.currentTime != null ? this.player.currentTime : 0,
         session: this.sessionDBId,
       };
-      EventAPIService.createEvent(eventData);
+      let response = await EventAPIService.createEvent(eventData);
+      if (eventType == "watching") this.watchingEventDBId = response.id;
+    },
+    /**
+     * Updates an event
+     * @param {String} eventType - The type of event that needs to be logged
+     * @param {Number} eventDBId - The id of an event as pulled from the database which needs to be updated
+     * @param {Object} eventDetails - An object with the details of the event
+     */
+    updateEvent(eventType, eventDBId, eventDetails = {}) {
+      if (!this.hasSessionStarted || !this.isAuthenticated || this.previewMode) return;
+      let eventData = {
```

Comment:
why create this variable?

Reply:
Just followed the style of the method `createEvent`
Also makes it a little more readable. Passing `eventData` to EventAPIService.udpateEvent()` feels more natural and readable.

If we'll get rid of this variable here, then we should do the same in `createEvent` method

Reply:
haan makes sense - let's do that